packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -166,19 +190,39 @@ export const SimpleFormIterator = (props: SimpleFormIteratorProps) => {
                         </SimpleFormIteratorItem>
                     ))}
                 </ul>
-                {!disabled && !disableAdd && (
-                    <div className={SimpleFormIteratorClasses.add}>
-                        {cloneElement(addButton, {
-                            className: clsx(
-                                'button-add',
-                                `button-add-${source}`
-                            ),
-                            onClick: handleAddButtonClick(
-                                addButton.props.onClick
-                            ),
-                        })}
-                    </div>
-                )}
+                <div className={SimpleFormIteratorClasses.line}>
+                    {!disabled && !disableAdd && (
+                        <div className={SimpleFormIteratorClasses.add}>
+                            {cloneElement(addButton, {
+                                className: clsx(
+                                    'button-add',
+                                    `button-add-${source}`
+                                ),
+                                onClick: handleAddButtonClick(
+                                    addButton.props.onClick
+                                ),
+                            })}
+                        </div>
+                    )}
+                    {fields.length > 0 && (
+                        <div className={SimpleFormIteratorClasses.clear}>
```

Comment:
Why create a different div than for the addButton? On Mobile, this will put the clear button in another line

![image](https://user-images.githubusercontent.com/99944/198593279-a08cc0b9-6345-42eb-8d70-b714f66b43dc.png)

Reply:
move into the same div with `addButton` 

commit 444eb59c612dc860dc24a70de2438043a86e086c